pkg/runtime/wfengine/activity.go

```diff
@@ -229,11 +234,7 @@ func (a *activityActor) loadActivityState(ctx context.Context, actorID string, g
 	result, ok := a.statesCache.Load(actorID)
 	if ok {
 		cachedState := result.(activityState)
-
-		// Make sure the cached state is for the same generation of the workflow.
-		if cachedState.Generation == generation {
```

Comment:
Now that we're not using the `Generation` field, let's delete it from `activityState`.